chirp/drivers/th_uv88.py

```diff
@@ -1094,6 +1261,496 @@ def get_settings(self):
+                      _optsignal.dtmf.group, True)
+
+        # Speed
+        options = ['50 ms', '100 ms', '200 ms', '300 ms', '500 ms']
+        rx = RadioSettingValueList(options, options[_optsignal.dtmf.speed])
+        rset = RadioSetting("optsignal.dtmf.speed", "Speed", rx)
+        dtmf.append(rset)
+
+        # First digit
+        _add_time(dtmf, "optsignal.dtmf.firstdigit",
+                  "First digit", 0, 0.1, 11, "%.1f s",
+                  _optsignal.dtmf.firstdigit)
+
+        # Autoreset time
+        _add_time(dtmf, "optsignal.dtmf.autoresettime",
+                  "Autoreset time", 0, 1, 26, "%d s",
+                  _optsignal.dtmf.autoresettime)
+
+        # ANI
+        rv = RadioSettingValueBoolean(_optsignal.dtmf.ani)
+        rx = RadioSetting("optsignal.dtmf.ani", "ANI", rv)
+        dtmf.append(rx)
+
+        # Stun
+        _add_hexlen(dtmf, "optsignal.dtmf.stuncode", "Stun",
+                    _optsignal.dtmf, "stuncode", 15,
+                    _optsignal.dtmf.stuncode,
+                    _optsignal.dtmf.stuncodelen)
+
+        # Kill
+        _add_hexlen(dtmf, "optsignal.dtmf.killcode", "Kill",
+                    _optsignal.dtmf, "killcode", 15,
+                    _optsignal.dtmf.killcode,
+                    _optsignal.dtmf.killcodelen)
+
+        dtmfenc = RadioSettingGroup("dtmfenc", "Encode")
+        dtmf.append(dtmfenc)
+
+        # Own ID
+        _add_hexlen(dtmfenc, "optsignal.dtmf.ownid", "Own ID",
+                    _optsignal.dtmf, "ownid", 8,
+                    _optsignal.dtmf.ownid,
+                    _optsignal.dtmf.ownidlen)
+
+        # PTT ID
+        self._add_pttid(dtmfenc, "optsignal.dtmf.pttid", "PTT ID",
+                        _optsignal.dtmf.pttid)
+
+        # PTT ID BEGIN
+        _add_hexlen(dtmfenc, "optsignal.dtmf.pttidbegin", "PTT ID Begin",
+                    _optsignal.dtmf, "pttidbegin", 24,
+                    _optsignal.dtmf.pttidbegin,
+                    _optsignal.dtmf.pttidbeginlen)
+
+        # PTT ID END
+        _add_hexlen(dtmfenc, "optsignal.dtmf.pttidend", "PTT ID End",
+                    _optsignal.dtmf, "pttidend", 24,
+                    _optsignal.dtmf.pttidend,
+                    _optsignal.dtmf.pttidendlen)
+
+        # Sidetone
+        rv = RadioSettingValueBoolean(_optsignal.dtmf.sidetone)
+        rx = RadioSetting("optsignal.dtmf.sidetone", "Sidetone", rv)
+        dtmfenc.append(rx)
+
+        # Select channel
+        options = ['1', '2', '3', '4', '5', '6', '7', '8',
+                   '9', '10', '11', '12', '13', '14', '15', '16']
+        rx = RadioSettingValueList(options, options[_settings.dtmfch])
+        rset = RadioSetting("basicsettings.dtmfch", "Select channel", rx)
+        dtmfenc.append(rset)
+
+        # Encode channels
+        for i in range(16):  # 0 - 15
+            sigchan = RadioSettingSubGroup("dtmfencchan%d" % i,
+                                           "Channel %02d" % (i+1))
```

Comment:
Nice use of SubGroup. This is new, I just added it recently, but I think it really helps organize things better. Note that the "FM radio" section of this radio could probably use some similar cleanup, if you feel like it :)

I wish the style checker would complain, but python conventional style is space around operators (so `i + 1`), with `=` being the exception when used in parameter lists.

Reply:
I found the SubGroup while looking at recent commits :) Yes FM settings would also benefit the use of subgroup. I will add this to the #11153 issue

just commited the style fix